### PR TITLE
Replace es-dev-server with @web/dev-server

### LINKED ISSUE #131: Replace es-dev-server with @web/dev-server
#### Is your feature request related to a problem? Please describe.

When it's ready, we should replace the `es-dev-server` with [`@web/dev-server`](https://github.com/modernweb-dev/web/tree/master/packages/dev-server).

### PR DESCRIPTION
**Pull request by abdonrd**
Fix #131

- [x] Replace es-dev-server with @web/dev-server
- [x] Use `esbuild` with the `@web/dev-server-esbuild` instead of `tsc` to watch compile
- [x] Update the build process

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -78,6 +78,6 @@
 
     <noscript>Please enable JavaScript to view this website.</noscript>
 
-    <script type="module" src="src-js/components/app-index.js"></script>
+    <script type="module" src="src/components/app-index.ts"></script>
```
Comment (abdonrd) [inline index.html:81]: If the `index.html` import the TypeScript file, the build process stops working...

**Comment by abdonrd** [comment]
It seems the current problem about this:

>[!] (plugin babel) TypeError: unknown: this.optionalCall is not a function

It's because the optional chaining using here:

https://github.com/IBM/pwa-lit-template/blob/df5cbbf9d3c0bc2768c484dfe2694a17083ce3ef/src/pages/page-not-found.ts#L38

https://github.com/IBM/pwa-lit-template/blob/df5cbbf9d3c0bc2768c484dfe2694a17083ce3ef/src/pages/page-not-found.ts#L46

**Comment by abdonrd** [comment]
It works if I update the `createSpaConfig.legacyBuild` to `false` 🤔

**Comment by abdonrd** [comment]
@LarsDenBakker do you know how can I debug [this error](https://github.com/IBM/pwa-lit-template/pull/138#issuecomment-689418336)? Thanks in advance!

**Comment by LarsDenBakker** [comment]
You can try updating esbuild, there was an issue with patterns like that in esbuild. 

I'm not sure if I'd recommend using esbuild for production build right now, sorry if I made that impression :) esbuild is very promising, but it still has bugs sometimes. Which is fine for development, but for production it might be problematic.

**Comment by abdonrd** [comment]
Got it! Thanks @LarsDenBakker!

**Comment by abdonrd** [comment]
Now we the migration to `@rollup/plugin-typescript`, we have a new [error](https://github.com/IBM/pwa-lit-template/pull/138/checks?check_run_id=1113035489#step:7:22)! 😁 

>(!) Plugin typescript: @rollup/plugin-typescript: Rollup 'sourcemap' option must be set to generate source maps.
>[!] (plugin typescript) Error: @rollup/plugin-typescript: 'outDir' or 'declarationDir' must be specified to generate declaration files.
>Error: @rollup/plugin-typescript: 'outDir' or 'declarationDir' must be specified to generate declaration files.

**Comment by abdonrd** [comment]
Now we the migration to `@rollup/plugin-typescript`, we have a new [error](https://github.com/IBM/pwa-lit-template/pull/138/checks?check_run_id=1118232667#step:7:14)! 😁 

>(!) Plugin typescript: @rollup/plugin-typescript: Rollup 'sourcemap' option must be set to generate source maps.
>[!] (plugin typescript) Error: @rollup/plugin-typescript: 'outDir' or 'declarationDir' must be specified to generate declaration files.
>Error: @rollup/plugin-typescript: 'outDir' or 'declarationDir' must be specified to generate declaration files.

**Comment by abdonrd** [comment]
Seems that the `@rollup/plugin-typescript` is working now! 🎉 

But we back to the [origin issue](https://github.com/IBM/pwa-lit-template/pull/138#issuecomment-689418336):

>[!] (plugin babel) TypeError: unknown: this.optionalCall is not a function

It's because these two lines because the optional calls:

https://github.com/IBM/pwa-lit-template/blob/df5cbbf9d3c0bc2768c484dfe2694a17083ce3ef/src/pages/page-not-found.ts#L38

https://github.com/IBM/pwa-lit-template/blob/df5cbbf9d3c0bc2768c484dfe2694a17083ce3ef/src/pages/page-not-found.ts#L46

And it works if I update the `createSpaConfig.legacyBuild` to `false` 🤔

**Comment by abdonrd** [comment]
For leaving a record, if I comment this line, it works:

https://github.com/open-wc/open-wc/blob/master/packages/building-rollup/src/createBasicConfig.js#L82

**Pull request merged**